app/assets/javascripts/editor.js.erb

```diff
@@ -404,12 +413,23 @@ $(function() {
       editor.setTheme(THEME);
       editor.commands.bindKey("ctrl+alt+0", null);
       editors.push(editor);
+      editor_for_file.set($(element).parent().data('filename'), editor);
       var session = editor.getSession();
       session.setMode($(element).data('mode'));
       session.setTabSize($(element).data('indent-size'));
       session.setUseSoftTabs(true);
       session.setUseWrapMode(true);
 
+      // set regex for parsing error traces based on the mode of the main file.
+      if( $(element).parent().data('role') == "main_file"){
+        var mode = $(element).data('mode');
+        if(mode == "ace/mode/python"){
+          tracepositions_regex = tracepositions_regexes.python;
+        } else if (mode = "ace/mode/java"){
+          tracepositions_regex = tracepositions_regexes.java;
+        }
+      }
+
```

Comment:
In case we add support for further programming languages this should be refactored to be used with a map.